internal/resolution/datasource/maven_registry.go

```diff
@@ -64,20 +78,25 @@ func (m *MavenRegistryAPIClient) WithoutRegistries() *MavenRegistryAPIClient {
 }
 
 // AddRegistry adds the given registry to the list of registries if it has not been added.
-func (m *MavenRegistryAPIClient) AddRegistry(registry string) error {
-	if slices.Contains(m.registries, registry) {
-		return nil
+func (m *MavenRegistryAPIClient) AddRegistry(registry MavenRegistry) error {
```

Comment:
Should this function be considering what happens when registries have the same `ID`?

Reply:
I just updated the code to use the ID to decide if the registry should be added, which matches Maven's behavior.